### DISCUSSION TITLE
How to define the physical parameters associated with variables

### DISCUSSION THREAD

**Discussion by AdelineHunter** [idaholab/moose#21656] [Q&A Modules: General] [answered]
Hi, Prof

I would like to ask how to define variable-related physical parameters in the input file.

For example, I want to define a density that is related to the temperature variable. Can I define the property parameters of this type through the `Function` in the input file without modifying the code.

Thanks.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Typically for a quantity that is dependent on another variable, we use the Materials and Material Properties system. 
Please see the documentation for that system here:
https://mooseframework.inl.gov/syntax/Materials/index.html

Guillaume

**Reply by AdelineHunter** [reply]
Thank you, Guillaume

Your advice is of great help to me.

I remember using a method that didn't need to change the code, just set the input file, but I seem to have forgotten it.

**Reply by GiudGiud** [reply]
there are parsed material expressions you may use:
https://mooseframework.inl.gov/source/materials/ParsedMaterial.html
and for fluid properties, it's easier to work with the fluid properties module to set all these quantities
https://mooseframework.inl.gov/modules/fluid_properties/index.html

**Reply by AdelineHunter** [reply]
Thank you, Guillaume

Thank you for your help.
